akka-actor/src/main/scala/akka/actor/ActorRef.scala

```diff
@@ -599,6 +599,16 @@ private[akka] class VirtualPathContainer(
 
   private val children = new ConcurrentHashMap[String, InternalActorRef]
 
+  override def !(message: Any)(implicit sender: ActorRef = Actor.noSender): Unit = message match {
+    case ActorSelectionMessage(msg, elements, _) ⇒ {
+      elements.foreach {
+        case SelectChildName(name)       ⇒ getChild(name).tell(msg, sender)
+        case SelectChildPattern(pattern) ⇒ // no idea
```

Comment:
This sould deliver the message to all children whose name matches the pattern. This is best effort, so no problem if the ConcurrentHashMap actually changes in-between.